source/socks5-support/tests/README.rst

```diff
@@ -0,0 +1,98 @@
+====================
+SOCKS5 Support Tests
+====================
+
+.. contents::
+
+----
+
+Introduction
+============
+
+This document describes how drivers should test support for SOCKS5 proxies.
+
+Testing Requirements
+====================
+
+For a single server version (at least 5.0), drivers MUST add two
+Evergreen tasks: one with a replica set with TLS enabled, and one
+with TLS disabled. The servers of the replica set may listen on any free ports.
+
+Each task MUST also start up two SOCKS5 proxies: one requiring authentication,
+and one that does not require authentication.
+
+SOCKS5 Proxy Configuration
+--------------------------
+
+Drivers MUST use the ``socks5srv.py`` script in ``drivers-evergreen-tools``
+to run the SOCKS5 proxy servers for Evergreen tasks. This script MUST
+be run after the backing replica set has already been started,
+and MUST be started with ``--map "localhost:12345 to <host>"` where
+``host` is the host:port identifier of an arbitrary member of the replica set.
+The SOCKS5 proxy server requiring authentication MUST be started with
+``--port 1080 --auth username:p4ssw0rd`, the one not requiring authentication
+with `--port 1081``.
+
+Prose Tests
```

Comment:
Instead of having these as prose tests, could we link to the URI options spec test readme (https://github.com/mongodb/specifications/blob/master/source/uri-options/uri-options.rst) and create spec tests in https://github.com/mongodb/specifications/tree/master/source/uri-options/tests instead?

Reply:
My understanding is that those tests only cover validity of the connection string, not actual connectivity? I’m happy to move to a more standard format, though, I was hoping that there would be a way to do that :)

Reply:
That’s correct. I ppppppppppppp

Reply:
Pppppppppppppppppppp

Reply:
I was looking at the prose tests and they seem to check validity of the options in the connection string, which is what the URI option spec tests check. I'm happy with connectivity tests as well, but I think the validation of the connection string options should reside in the same place as the others. Thoughts?

Reply:
So just to be clear – you’d like to move the invalid connection strings over to the URI option spec, and leave the rest as-is? I think that sounds okay, it would even still allow drivers to bump the connection string spec without having to actually implement proxy support.

Reply:
Yes exactly. Thanks!

Reply:
@durran Sure, done!